lib/team_api/joiner.rb

```diff
@@ -75,7 +75,18 @@ def join_project_data
       # convention takes hold, hopefully.
       projects = (data['projects'] ||= {})
       projects.delete_if { |_, p| p['status'] == 'Hold' } if @public_mode
-      projects.values.each { |p| join_team_list p['team'] }
+      projects.values.each do |p| 
+        if p['errors'].nil?
```

Comment:
This `if p['errors'].nil?` block can be elided as `p['errors'] ||= []`.

Conceivably, you could even elide it into: `join_team_list p['team'], (p['errors'] ||= [])`.

Or, create `errors = []`, then replace the final block with `p['errors'] = errors unless errors.empty?`. (Some folks don't like `unless`, but I find it perfectly readable in this context.)